api/controllers/game/stalemate.js

```diff
@@ -0,0 +1,56 @@
+module.exports = async function (req, res) {
+  try {
+    const game = await Game.findOne({ id: req.session.game }).populate('players');
+    let gameUpdates = {};
+    const updatePromises = [];
+
+    switch (req.session.pNum) {
+      case 0: {
+        const newVal = game.turnStalemateWasRequestedByP0 ? game.turn : null;
+        gameUpdates.turnStalemateWasRequestedByP0 = newVal;
+        game.turnStalemateWasRequestedByP0 = newVal;
+        break;
+      }
+      case 1: {
+        const newVal = game.turnStalemateWasRequestedByP1 ? game.turn : null;
+        gameUpdates.turnStalemateWasRequestedByP1 = newVal;
+        game.turnStalemateWasRequestedByP1 = newVal;
+        break;
+      }
+    }
```

Comment:
Switch works here, but you could probably also DRY up the logic a bit.
```suggestion
    const { pNum } = req.session;
    const turnKey = `turnStalemateWasRequestedByP${pNum}`;
    const newVal = game[turnKey] ? game.turn : null;
    gameUpdates[turnKey] = newVal;
    game[turnKey] = newVal;
```

Reply:
You'd also need to update the below code, so don't just add this. It's strictly a suggestion for code style.